src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -90,138 +92,151 @@ class FastSync(
         syncController ! Done
       } else {
         val initialSyncState =
-          SyncState(targetBlockHeader, safeDownloadTarget = targetBlockHeader.number + syncConfig.fastSyncBlockValidationX)
+          SyncState(targetBlock = targetBlockHeader, safeDownloadTarget = targetBlockHeader.number + syncConfig.fastSyncBlockValidationX)
         startWithState(initialSyncState)
       }
   }
 
   // scalastyle:off number.of.methods
-  private class SyncingHandler(initialSyncState: SyncState) {
-
-    private val BlockHeadersHandlerName = "block-headers-request-handler"
+  private[sync] class SyncingHandler(initialSyncState: SyncState) {
 
-    private var requestedHeaders: Map[Peer, BigInt] = Map.empty
-
-    private var syncState = initialSyncState
-
-    private var assignedHandlers: Map[ActorRef, Peer] = Map.empty
-    private var peerRequestsTime: Map[Peer, Instant] = Map.empty
-
-    private var requestedMptNodes: Map[ActorRef, Seq[HashType]] = Map.empty
-    private var requestedNonMptNodes: Map[ActorRef, Seq[HashType]] = Map.empty
-    private var requestedBlockBodies: Map[ActorRef, Seq[ByteString]] = Map.empty
-    private var requestedReceipts: Map[ActorRef, Seq[ByteString]] = Map.empty
-
-    private val syncStateStorageActor = context.actorOf(Props[FastSyncStateStorageActor], "state-storage")
+    private val syncStateStorageActor = context.actorOf(Props[FastSyncStateStorageActor], StateStorageName)
     syncStateStorageActor ! fastSyncStateStorage
 
-    //Delay before starting to persist snapshot. It should be 0, as the presence of it marks that fast sync was started
+    // Delay before starting to persist snapshot. It should be 0, as the presence of it marks that fast sync was started
     private val persistStateSnapshotDelay: FiniteDuration = 0.seconds
     private val syncStatePersistCancellable = scheduler.schedule(persistStateSnapshotDelay, persistStateSnapshotInterval, self, PersistSyncState)
     private val printStatusCancellable = scheduler.schedule(printStatusInterval, printStatusInterval, self, PrintStatus)
     private val heartBeat = scheduler.schedule(syncRetryInterval, syncRetryInterval * 2, self, ProcessSyncing)
 
-    def receive: Receive = handleCommonMessages orElse {
-      case UpdateTargetBlock(state) => updateTargetBlock(state)
-      case ProcessSyncing => processSyncing()
-      case PrintStatus => printStatus()
-      case PersistSyncState => persistSyncState()
+    def receive(handlerState: HandlerState): Receive =
+      handleCommonMessages orElse
+      handleSyncing(handlerState) orElse
+      handleReceivedResponses(handlerState) orElse {
+      case UpdateTargetBlock(state) =>
+        updateTargetBlock(state, handlerState)
+
+      case PeerRequestHandler.RequestFailed(peer, reason) =>
```

Comment:
Minor: i would prefer to have it in `handleReceivedResponses` as it is nice to have failure case neat all other responses